### ISSUE TITLE
Shouldn't it refer to layers with a unique handle rather than by name?

### ISSUE THREAD

**Issue by AlisterH** [REDcatch/Volume_calculation_for_QGIS3#8]
If a project has two vector layers with the same name, this plugin lists both of them, but gives the same results regardless of which one is selected.
I guess the code should be changed to refer to the layer by its unique handle rather than by name.
I didn't test what happens if you have two different raster layers with the same name.

**Comment by distbug** [comment]
That is true, however, we do or did not think that this is a common use case. Is this a common use case for you ? We could change it, but we'd need more demand for it first.

**Comment by AlisterH** [comment]
I can only speak for myself, but I almost always have multiple layers with the same name, particularly raster layers.

This is because I tend to work with a lot of temporary layers created by various processing algorithms, so they all just have default names like OUTPUT.  Eventually I save permanent copies of some of the layers, but I don't want to keep most of them, so it is easier to create them as temporary layers than it is to initially save them all as 'permanent' outputs and then go through later manually deleting the ones I don't need.

I guess in terms of this plugin the simple use case is something like this:
- create a layer
- calculate volumes
- create a slightly different layer
- calculate volumes.  If the two layers have the same name it shouldn't really matter - it would be obvious if I didn't select the new layer because I would get the same answer again, so I could then select it and recalculate.

**Comment by distbug** [comment]
If you want you can open a pull-request for this change. As far as I can see this would mostly involve switching out the calls for the map layers while presenting the layer name to the user with its unique handle also being tracked/saved.
